aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -723,8 +723,7 @@ constexpr double QUARTER = 0.25;
 }
 
 static inline void mvlgamma_check(const Tensor& self, int64_t p) {
-  TORCH_CHECK((self > HALF * (p - 1)).all().item<bool>(),
-              "All elements must be greater than (p-1)/2");
+  TORCH_CHECK(self.scalar_type() != kBool, "The input tensor may not be a boolean tensor.");
```

Comment:
this looks like a different check than what used to be here?

Reply:
As per the OP:
>I removed the data-dependent check for the args, as data dependencies
are bad for many reasons (and it was failing when the input has NaNs).

The new check is because some test was failing for bools, so I simply disallowed bools for this function, as they don't make any sense anyway.